cluster-diagnosis/sysdumpcollector.py

```diff
@@ -120,23 +120,45 @@ def collect_logs_per_pod(self, podstatus):
         else:
             log.info("collected log file: {}".format(log_file_name))
 
-        # Previous containers
-        log_file_name_previous = "{0}-previous".format(log_file_name)
-        cmd = command.format(
-            "--previous", self.since, self.size_limit, podstatus[4],
-            podstatus[0],
-            self.sysdump_dir_name, log_file_name_previous)
+        # Only get logs from previous containers if pod has restarted.
+        # Need to get the pod to access its restartCount.
+        command = "kubectl get pod {} -n {} -o json"
+        podCmd = command.format(
+            podstatus[0], podstatus[4]
+        )
+
         try:
-            subprocess.check_output(cmd, shell=True)
+            podOutput = subprocess.check_output(podCmd, shell=True)
         except subprocess.CalledProcessError as exc:
             if exc.returncode != 0:
-                log.debug(
-                    "Debug: {}. Could not collect previous "
-                    "log for '{}': {}"
-                    .format(exc, podstatus[0], log_file_name))
+                log.debug("Debug {}: could not get pod {}").format(
+                    exc, podstatus[0])
         else:
-            log.info("collected log file: {}".format(
-                log_file_name_previous))
+            # Examine JSON output to see if restartCount > 0
+            decodedPodOutput = podOutput.decode()
+            jsonOutput = json.loads(decodedPodOutput)
+            containerStatus = jsonOutput['status']['containerStatuses']
+            restartCount = containerStatus[0]['restartCount']
```

Comment:
There can be multiple containers in a pod. I think this code assumes we care about only the first one. That's ok for now, but please add a TODO comment.

Reply:
Fixed